accounts/abi/bind/bind.go

```diff
@@ -47,7 +47,7 @@ const (
 // to be used as is in client code, but rather as an intermediate struct which
 // enforces compile time type safety and naming convention opposed to having to
 // manually maintain hard coded strings that break on runtime.
-func Bind(types []string, abis []string, bytecodes []string, fsigs []map[string]string, pkg string, lang Lang, libs map[string]string, aliases map[string]string) (string, error) {
+func Bind(types []string, abis []string, bytecodes []string, fsigs []map[string]string, pkg string, lang Lang, libs map[string]string, aliases map[string]string, opts ...BindOption) (string, error) {
```

Comment:
Thinking aloud here, without having reviewed/understood the entire method, but.. 
This is an api-breaking change, so would it make sense to instead add a different method
`BindWithOpts( x, y, z,  .. , opts ... BindOption) (stnrg, error)` instead? 

And possible bundle as much as possible of the _existing_ input arguments as `BindOption`s.

Reply:
It's actually non-breaking by design. The variadic `...` allows the function to be called with zero or more individual `BindOption`s so all existing code remains unaffected—I think there's at least one call to `Bind()` in the Java tests that remains unchanged.

This is a common pattern in, for example, Google's gRPC https://pkg.go.dev/google.golang.org/grpc